src/layer/vector/Canvas.js

```diff
@@ -284,13 +285,34 @@ export const Canvas = Renderer.extend({
 
 		ctx.beginPath();
 
-		for (i = 0; i < len; i++) {
-			for (j = 0, len2 = parts[i].length; j < len2; j++) {
-				p = parts[i][j];
-				ctx[j ? 'lineTo' : 'moveTo'](p.x, p.y);
+		if (layer.options.curved) {
+			for (let i = 0; i < parts.length; i++) {
+				const points = parts[i];
+
+				const cmds = curvedPathCommands(points, closed);
+				for (j = 0; j < cmds.length; j++) {
+					const cmd = cmds[j];
+					switch (cmd.type) {
```

Comment:
I can't see the point of holding "command" objects. The non-bezier code can differentiate between `lineTo` and `moveTo` by `j===0`, so there is no need to do an initial `array.push` of a `M` "command" and then push `C` commands.

Reply:
I don't understand what you mean.
Do you want me to replace `curvedPathCommands` with in-loop calculation of Bezier curve parameters?
It would be more efficient but then the code in the loop would become more complicated.
Also, for polygons (or polylines with `closed=true`) `curvedPathCommands` modifies the first `C` command at the end of the loop to make the joint smooth. It would not be possible if `curvedPathCommands` was extracted here.